finetune/nn/auxiliary.py

```diff
@@ -1,21 +1,159 @@
+import math
+
 import tensorflow as tf	
 from finetune.util.shapes import shape_list
+from finetune.util.positional_embeddings import add_timing_signal_from_position
+from finetune.base_models.gpt.featurizer import conv1d
+
+def layer_norm_with_custom_init(input_tensor, begin_norm_axis=-1, begin_params_axis=-1,
+                                name=None, custom=False, pos_embed=None):
+    """Run layer normalization on the last dimension of the tensor."""
+
+    if custom:
+        bert_dimension = shape_list(input_tensor)[1] - pos_embed
+
+        bert_tensor = input_tensor[:,:bert_dimension]
```

Comment:
I'm not sure we can be certain that the 2nd dimension is always the feature dimension can we?

Reply:
Have verified, Apparently we can.